### ISSUE TITLE
Target flow

### ISSUE THREAD

**Issue by nonken** [nonken/lambda-cdk#1]
Goal: 

Run a CDK command which:

- Creates pipeline and ApiGateway.
- Iterates over `src` directory and:
- Create API Gateway routes. (right now this is hard-coded)
- Creates build artifact specific to the route.

- Creates deployment specific to the rouce.
- Creates pipeline and deployments for Lambdas based on the handlers in `src`.

Issues:

**Comment by nonken** [comment]
I am getting this dependency error atm:

```Error: 'PipelineDeployingLambdaStack' depends on 'EndpointStack' (PipelineDeployingLambdaStack -> EndpointStack/LambdaBuild/Role/Resource.Arn, PipelineDeployingLambdaStack -> EndpointStack/LambdaBuild/Resource.Ref, PipelineDeployingLambdaStack -> EndpointStack/LambdaBuild/Resource.Arn). Adding this dependency (EndpointStack -> PipelineDeployingLambdaStack/Pipeline/ArtifactsBucket/Resource.Arn) would create a cyclic reference.```

**Comment by rix0rrr** [comment]
Something inside `PipelineDeployingLambdaStack` seems to be referencing something called `LambdaBuild` in `EndpointStack`.

Something inside `EndpointStack` seems to be referencing the artifacts bucket that is in `PipelineDeployingLambdaStack`.

If `LambdaBuild` is a CodeBuild project, to me it would seem to belong in the PipelineStack, not in the Deployed Stack. Is that it?

**Comment by nonken** [comment]
Yeah indeed, I tried to move what I thought was `PipelineDeployingLambdaStack/Pipeline/ArtifactsBucket/Resource.Arn` into a shared stack which is consumed by both but thats not it... Digging in more.

https://github.com/nonken/lambda-cdk/blob/master/infra/shared.js

**Comment by nonken** [comment]
I will try and move the pipeline actions back to the pipeline stack and see if that works. I think I tried that and also had a weird cyclical dependency issue 🤣
